lib/apt/spy2.rb

```diff
@@ -121,6 +121,42 @@ def filter(mirrors, strict = false, output = true)
 
   end
 
+  private
+  # architecture is not reported in the strings we need
+  def fix_arch(str)
+    if str == "x86_64"
+      return "amd64"
+    end
+
```

Comment:
@till We should also add check for `arm64`.

Reply:
Do you happen to know where to look up possible strings? Or do you have access to a VM? I'll google otherwise later.

Reply:
I think I found a better solution. LMK

Reply:
`dpkg` is the initial thing I tried but the machine I was checking on had Amazon Linux Image installed, where `dpkg` was missing! We could default to returning `mirror` from the method  `get_mirror()` if `get_arch()` fails! Let me know what you think!

Reply:
On second thought, I think this is fine. The user could always not use the `--strict` flag 😸 
Thanks for fixing this, now we can rely on this tool.

Reply:
Amazon Linux doesn't have it because it's CentOS/Fedora based and uses yum. This would only work for Ubuntu.